main.go

```diff
@@ -502,7 +507,26 @@ func (c *controller) sync() {
 			continue
 		}
 
+		// If there's an increase in replicas we poll for the new replicas to be ready
+		if _, ok := c.replicas[hashring]; ok && c.replicas[hashring] < *sts.Spec.Replicas {
```

Comment:
For this type of thing to be truly safe to do, we'll need to eventually implement leader election to avoid that multiple controllers are reconciling this state.

Reply:
I'm not aware that anyone has tried running multiple thanos-receive-controllers yet. If we are going to support that, then yes, that should be improved. :)

Reply:
Agreed with Matthias. Right now, it's definitely assumed that the controller is only one replica. Even without this change, we would really need to implement some coordination to enable more replicas to have truly predictable results.

Reply:
while unlikely, the case that I worry about more is rollout/eviction/preemption or some other reason why there may be temporarily two .. as I said, _eventually_ this would be good, not necessary in this PR